src/components/RolesTable/index.jsx

```diff
@@ -0,0 +1,124 @@
+import { Fragment, Component } from 'react';
+import { Link } from 'react-router-dom';
+import { string, arrayOf } from 'prop-types';
+import {
+  memoizeWith,
+  pipe,
+  prop,
+  map,
+  ifElse,
+  isEmpty,
+  filter,
+  contains,
+  identity,
+  sort as rSort,
+} from 'ramda';
+import { camelCase } from 'change-case/change-case';
+import { TableRow, TableCell } from 'material-ui/Table';
+import Typography from 'material-ui/Typography';
+import { ListItemText } from 'material-ui/List';
+import LinkIcon from 'mdi-react/LinkIcon';
+import TableCellListItem from '../TableCellListItem';
+import { role } from '../../utils/prop-types';
+import sort from '../../utils/sort';
+import DataTable from '../DataTable';
+
+const sorted = pipe(
+  rSort((a, b) => sort(a.roleId, b.roleId)),
+  map(({ roleId }) => roleId)
+);
+
+export default class RolesTable extends Component {
+  static propTypes = {
+    /** A GraphQL roles response. */
+    roles: arrayOf(role).isRequired,
+    /** A search term to refine the list of roles */
+    searchTerm: string,
+  };
+
+  static defaultProps = {
+    searchTerm: null,
+  };
+
+  state = {
+    sortBy: null,
+    sortDirection: null,
+  };
+
+  handleHeaderClick = sortBy => {
+    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
+    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';
+
+    this.setState({ sortBy, sortDirection });
+  };
+
+  createSortedRoles = memoizeWith(
+    (roles, sortBy, sortDirection, searchTerm) => {
+      const ids = sorted(roles);
+
+      return `${ids.join('-')}-${sortBy}-${sortDirection}-${searchTerm}`;
+    },
+    (roles, sortBy, sortDirection, searchTerm) => {
+      const sortByProperty = camelCase(sortBy);
+      const filteredRoles = ifElse(
```

Comment:
If we can't be fully point-free here, there's not much benefit to using curried functions.